engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/impl/RangePredicateFieldSetContextImpl.java

```diff
@@ -141,7 +147,12 @@ N below(Object value, RangeBoundInclusion inclusion) {
 					break;
 			}
 			checkHasNonNullBound();
-			return getNextContextProvider().get();
+			return new ExplicitEndContext<N>() {
```

Comment:
Same as above: you could just make `RangePredicateFieldSetContextImpl` implement `ExplicitEndContext` and return `this` here. Like we did in other places.
Or, even better, you could make `MultiFieldPredicateCommonState` implement `ExplicitEndContext`, and just return the common state here.